### PR TITLE
Pdb reaper metrics

### PR DESCRIPTION
**Pull request by shaoxt**
Push PDB reaper metrics to prometheus pushgateway

### CODE REVIEW

**Comment by shaoxt** [comment]
Feature verified:

```
governor_pdb_reaper_result{container="prometheus-pushgateway", endpoint="http", job="governor", namespace="hpa-samples", pdb="hpa-samples-pdb", pod="prometheus-pushgateway-6999dbd5c7-hdc84", reason="BlockingPodDisruptionBudget", service="prometheus-pushgateway"} | 0
-- | --
governor_pdb_reaper_result{container="prometheus-pushgateway", endpoint="http", job="governor", namespace="hpa-samples", pdb="hpa-samples-pdb", pod="prometheus-pushgateway-6999dbd5c7-hdc84", reason="BlockingPodDisruptionBudgetWithCrashLoop", service="prometheus-pushgateway"} | 0
governor_pdb_reaper_result{container="prometheus-pushgateway", endpoint="http", job="governor", namespace="hpa-samples", pdb="hpa-samples-pdb", pod="prometheus-pushgateway-6999dbd5c7-hdc84", reason="BlockingPodDisruptionBudgetWithNotReadyState", service="prometheus-pushgateway"} | 0
governor_pdb_reaper_result{container="prometheus-pushgateway", endpoint="http", job="governor", namespace="hpa-samples", pdb="hpa-samples-pdb", pod="prometheus-pushgateway-6999dbd5c7-hdc84", reason="PodDisruptionBudgetDeleted", service="prometheus-pushgateway"}

```

File: pkg/reaper/pdbreaper/pdbreaper.go [hunk 1]
```diff
@@ -146,6 +151,8 @@ func (ctx *ReaperContext) scan() error {
 		}
 
 		ctx.ClusterBlockingPodDisruptionBudgets[namespace] = append(ctx.ClusterBlockingPodDisruptionBudgets[namespace], pdb)
+
+		ctx.exposeMetric(pdb, EventReasonPodDisruptionBudgetDeleted, 0)
```
Comment (eytan-avisror) [inline pkg/reaper/pdbreaper/pdbreaper.go:155]: This is prior to actual deletion

Comment (eytan-avisror) [reply pkg/reaper/pdbreaper/pdbreaper.go:155]: If this is just initialization, can we put this higher up even prior to scanning?

**Review by eytan-avisror — COMMENTED** [review]
LG, did you get a chance to test these changes out?

File: pkg/reaper/pdbreaper/pdbreaper.go (hunk 1 above)
Comment (shaoxt) [reply pkg/reaper/pdbreaper/pdbreaper.go:155]: 0 means no deletion.  Just to set the initial value of the metric

**Comment by shaoxt** [comment]
> LG, did you get a chance to test these changes out?

Yes,  
Here are the metrics

```
governor_pdb_reaper_result{container="prometheus-pushgateway", endpoint="http", job="governor", namespace="hpa-samples", pdb="hpa-samples-pdb", pod="prometheus-pushgateway-6999dbd5c7-hdc84", reason="BlockingPodDisruptionBudget", service="prometheus-pushgateway"} | 0
-- | --
governor_pdb_reaper_result{container="prometheus-pushgateway", endpoint="http", job="governor", namespace="hpa-samples", pdb="hpa-samples-pdb", pod="prometheus-pushgateway-6999dbd5c7-hdc84", reason="BlockingPodDisruptionBudgetWithCrashLoop", service="prometheus-pushgateway"} | 0
governor_pdb_reaper_result{container="prometheus-pushgateway", endpoint="http", job="governor", namespace="hpa-samples", pdb="hpa-samples-pdb", pod="prometheus-pushgateway-6999dbd5c7-hdc84", reason="BlockingPodDisruptionBudgetWithNotReadyState", service="prometheus-pushgateway"} | 0
governor_pdb_reaper_result{container="prometheus-pushgateway", endpoint="http", job="governor", namespace="hpa-samples", pdb="hpa-samples-pdb", pod="prometheus-pushgateway-6999dbd5c7-hdc84", reason="PodDisruptionBudgetDeleted", service="prometheus-pushgateway"}
```

**Comment by eytan-avisror** [comment]
@shaoxt you need to sign the commit per DCO

**Review by eytan-avisror — APPROVED** [review]

**Pull request closed without merging**